ansible/roles/products/hand-e/server/desktop-icons/tasks/main.yml

```diff
@@ -0,0 +1,133 @@
+    launch_terminal: "true"
+
+- name: Install desktop icon for launching ROSCORE
+  import_tasks: default-icon.yml
+  vars:
+    template: ../../../common/resources/templates/scripts/start-roscore.j2
+    desktop_icon_png: "ROS_logo.png"
+    launch_script: "shadow_roscore.sh"
+    desktop_icon_name: "2 - Launch Server ROSCORE"
+    desktop_icon_path: "Shadow Advanced Launchers/2 - Launch Server ROSCORE"
+    launch_terminal: "true"
+
+- name: Install desktop icon for NUC hardware control loop
+  import_tasks: hand-icon.yml
+  vars:
+    desktop_icon_png: "hand-e.png"
+    launch_script: "shadow_nuc_hardware_control_loop.sh"
+    hand_branch: "{{ config_branch }}"
+    desktop_icon_name: "3 - Launch NUC Container and Hardware Control Loop"
+    desktop_icon_path: "Shadow Advanced Launchers/3 - Launch NUC Container and Hardware Control Loop"
+    ethercat_interface_var: "{{ ethercat_interface }}"
+
+- name: Install desktop icon for hand GUI
+  import_tasks: roslaunch-icon.yml
+  vars:
+    desktop_icon_png: "rviz.png"
+    launch_script: "shadow_GUI.sh"
+    project_name_input: "sr_robot_launch"
+    launch_file_input: "srhand.launch external_control_loop:=true sim:=false"
+    desktop_icon_name: "4 - Launch Server GUI"
+    desktop_icon_path: "Shadow Advanced Launchers/4 - Launch Server GUI"
+
+- name: Install desktop icon for NUC container
+  include_tasks: default-icon.yml
+  vars:
+    template: templates/scripts/start-docker-container-only-nuc.j2
+    desktop_icon_png: "nuc.jpg"
+    launch_script: "shadow_nuc_container.sh"
+    desktop_icon_name: "Launch NUC Container"
+    desktop_icon_path: "Shadow Advanced Launchers/Launch NUC Container"
+    command_to_run: "bash"
+    launch_terminal: "true"
+
+- name: Include products/common/demo-icons role
+  include_role:
+    name: products/common/demo-icons
+  vars:
+    demo_icon_folder: "{{ user_folder }}/Desktop/Shadow Demos"
+  when: demo_icons|bool
+
+- name: Copy Launch Local Shadow Hand desktop icon
+  copy:
+    src: files/hand-e.png
+    dest: "{{ shadow_hand_launcher_folder }}/hand-e.png"
+    mode: '664'
+
+- name: Create Launch Local Shadow Hand script
+  template:
+    src: ../../../common/resources/templates/scripts/start-docker-container.j2
+    dest: "{{ shadow_hand_launcher_folder }}/shadow_local_launcher_exec.sh"
+    mode: '755'
+
+- name: Create Shadow Local Hand Launch desktop icon
+  template:
+    src: ../../../common/resources/templates/desktop-icons/standard-icon.j2
+    dest: "{{ user_folder }}/Desktop/Shadow Advanced Launchers/Launch Local Shadow Hand.desktop"
+    mode: '755'
+  vars:
+    desktop_shortcut_name: "Launch Local Shadow Hand"
+    comment: "This is application launches the hand locally without the NUC"
+    folder: "{{ shadow_hand_launcher_folder }}"
+    shell_script_file_name: shadow_local_launcher_exec.sh
+    icon_file_name: hand-e.png
+    start_terminal: "true"
+
+- name: Make Shadow Local Hand Launch desktop icon Trusted
```

Comment:
shouldnt we use this for every icon?

Reply:
We are using it for every icon. All the other icons are using default-icon.yml, hand-icon.yml, roslaunch-icon.yml, which all include making the desktop icon trusted. No one has reported any desktop icon not working in 18.04. I think they are all fine. If you know some icon is not trusted, let me know.